docs/book/src/migration/v3vsv4.md

```diff
@@ -0,0 +1,84 @@
+# go/v3 vs go/v4-alpha 
+
+This document covers all breaking changes when migrating from v2 to v3.
+
+The details of all changes (breaking or otherwise) can be found in
```

Comment:
Nits:

```
The details of all changes (breaking or otherwise) can be found in:
- [controller-runtime][controller-runtime]
- [controller-tools][controller-tools]
- [kustomize][kustomize-release]
- [kb-releases][kb-releases] release notes.
```